Mathlib/Control/Monad/Cont.lean

```diff
@@ -120,6 +120,20 @@ instance (ε) [MonadExcept ε m] : MonadExcept ε (ContT r m) where
   throw e _ := throw e
   tryCatch act h f := tryCatch (act f) fun e => h e f
 
+instance (ε) [MonadExceptOf ε m] : MonadExceptOf ε (ContT r m) where
+  throw e _ := throw e
+  tryCatch act h f := tryCatch (act f) fun e => h e f
+
+instance (r m) [Monad m] [Lean.AddErrorMessageContext m] :
+    Lean.AddErrorMessageContext (ContT r m) where
+  add stx msg := monadLift <| Lean.AddErrorMessageContext.add stx msg
+
+instance (r m) [Monad m] [Lean.MonadRef m] : Lean.MonadRef (ContT r m) where
+  getRef := monadLift <| Lean.MonadRef.getRef
+  withRef stx h := Lean.MonadRef.withRef stx ∘ h
```

Comment:
This one I'm not certain about.

Reply:
This is tricky. We want to make sure that the continuation gets the old ref, otherwise `withRef` leaks the ref out of `h`.

I think this might correct, and it should handle exceptions too:

```lean
  withRef stx h := fun f => do
    let oldStx ← Lean.MonadRef.getRef
    Lean.MonadRef.withRef stx (h <| Lean.MonadRef.withRef oldStx ∘ f)
```